parsing/extend.ml

```diff
@@ -62,11 +62,11 @@ type simple_constr_prod_entry_key =
 
 (** Entries used in productions (in right-hand-side of grammar rules), to parse non-terminals *)
 
-type binder_entry_kind = ETBinderOpen | ETBinderClosed of (bool * string) list
-
 type binder_target = ForBinder | ForTerm
 
-type constr_prod_entry_key =
+type binder_entry_kind = ETBinderOpen | ETBinderClosed of constr_prod_entry_key option * (bool * string) list
+
+and constr_prod_entry_key =
```

Comment:
I'd rather move it upward and use `type` instead of `and` since there is no mutual recursivity (but I agree this enlarge the diff a bit).

Reply:
There is a recursivity (in `ETProdBinderList`).

Reply:
Indeed, my bad